File: android-client/data/src/main/java/ru/hits/hubabank/data/network/credit/model/CreditTermsDto.kt
package ru.hits.hubabank.data.network.credit.model

import kotlinx.serialization.Serializable
import ru.hits.hubabank.domain.credit.model.CreditTerms

@Serializable
internal data class CreditTermsDto(
    val id: String,
    val interestRate: Float,
    val title: String?,
    val isDeleted: Boolean,
)

internal fun CreditTermsDto.toDomain(): CreditTerms {
    return CreditTerms(
        id = id,
        interestRate = interestRate,
        title = title,
    )
}

File: BFF-client/BFF-client.Api/Models/Credit/CreditTermsDto.cs
namespace BFF_client.Api.Models.Credit
{
    public class CreditTermsDto
    {
        public Guid Id { get; set; }

        public float InterestRate { get; set; }

        public string title { get; set; }

        public bool IsDeleted { get; set; }
    }
}
